git-multimail/git_multimail.py

```diff
@@ -1643,17 +1643,41 @@ def send(self, lines, to_addrs):
 class SMTPMailer(Mailer):
     """Send emails using Python's smtplib."""
 
-    def __init__(self, envelopesender, smtpserver):
+    def __init__(self, envelopesender, smtpserver, 
+            smtpservertimeout = 10, smtpserverdebug = 0,
+            smtpserversecurity = 'none',
+            smtpserverusername = '', smtpserverpassword = '',
+            ):
         if not envelopesender:
             sys.stderr.write(
                 'fatal: git_multimail: cannot use SMTPMailer without a sender address.\n'
                 'please set either multimailhook.envelopeSender or user.email\n'
                 )
             sys.exit(1)
+        if smtpserversecurity == 'ssl' and not (smtpserverusername and smtpserverpassword):
+            sys.stderr.write(
+                'fatal: git_multimail: cannot use SMTPMailer with security option sll\n'
+                'without options username and password.\n'
+                )
+            sys.exit(1)
```

Comment:
Here and below: wouldn't it be better to throw an exception here, to let the caller possibly catch it (when using git_multimail.py as a Python module)?